tuf/client/updater.py

```diff
@@ -650,89 +650,73 @@ def refresh(self, unsafely_update_root_if_necessary=True):
     # _update_metadata() calls below do NOT perform an update if there
     # is insufficient trusted signatures for the specified metadata.
     # Raise 'tuf.NoWorkingMirrorError' if an update fails.
-
-    # Is the Root role expired?  When the top-level roles are initially loaded
-    # from disk, their expiration is not checked to allow their updating when
-    # requested (and give the updater the chance to continue, rather than always
-    # failing with an expired metadata error.)  If
-    # 'unsafely_update_root_if_necessary' is True, update an expired Root role
-    # now.  Updating the other top-level roles, regardless of their validity,
-    # should only occur if the root of trust is up-to-date.
     root_metadata = self.metadata['current']['root']
+    
     try: 
       self._ensure_not_expired(root_metadata, 'root')
-    
     except tuf.ExpiredMetadataError:
       # Raise 'tuf.NoWorkingMirrorError' if a valid (not expired, properly
       # signed, and valid metadata) 'root.json' cannot be installed.
       if unsafely_update_root_if_necessary:
         message = \
           'Expired Root metadata was loaded from disk.  Try to update it now.' 
         logger.info(message)
-        self._update_metadata('root', DEFAULT_ROOT_UPPERLENGTH)
      
       # The caller explicitly requested not to unsafely fetch an expired Root.
       else:
         logger.info('An expired Root metadata was loaded and must be updated.')
         raise
 
-    # If an exception is raised during the metadata update attempts, we will
-    # attempt to update root metadata once by recursing with a special argument
-    # (unsafely_update_root_if_necessary) to avoid further recursion.
 
+    self._update_root_metadata(root_metadata)
```

Comment:
This is a major change to the update process.  Ideally, this sort of change should be mentioned in the PR summary and reflected in the proposal and TUF specification.

The docstring should also be updated to state that root.json is now always updated before the other top-level roles.  I'll take care of updating the documentation.  I was debugging a problem and was confused as to why the update process was behaving this way.  Hadn't noticed this one line change to refresh().